ui/webui/src/components/Error.jsx

```diff
@@ -16,43 +16,128 @@
+};
+
+const addExceptionDataToReportURL = (url, exception) => {
+    const newUrl = new URL(url);
+    newUrl.searchParams.append(
+        "short_desc",
+        "WebUI: " + exception.name + ": " + exception.message
+    );
+    newUrl.searchParams.append(
+        "comment",
+        "Installer WebUI Critical Error:\n" + exception.name + ": " + exception.message + "\n\n" + _("Please attach the file /tmp/webui.log to the issue.")
+    );
+    return newUrl.href;
+};
+
+export const CriticalError = ({ exception, isBootIso, reportLinkURL }) => {
+    const reportURL = addExceptionDataToReportURL(reportLinkURL, exception);
+    const [logContent, setLogContent] = useState("");
+
+    useEffect(() => {
+        cockpit.spawn(["journalctl", "-a"])
+                .then(content => setLogContent(content));
+    }, []);
+
+    const openBZIssue = (reportURL) => {
+        cockpit.file("/tmp/webui.log").replace(logContent)
+                .then(window.open(reportURL, "_blank", "noopener,noreferer"));
+    };
+
+    const context = exception.contextData?.context;
+
     return (
         <Modal
-          description={_("The installer cannot continue due to a critical error.")}
+          description={context
+              ? cockpit.format(_("The installer cannot continue due to a critical error: $0"), context)
+              : _("The installer cannot continue due to a critical error.")}
           id="critical-error-modal"
           isOpen
           position="top"
           showClose={false}
           title={_("Critical error")}
           titleIconVariant="danger"
-          variant="small"
+          variant={ModalVariant.large}
           footer={
               <>
+                  {reportLinkURL &&
+                  <Button
+                    variant="primary"
+                    icon={<ExternalLinkAltIcon />}
+                    onClick={() => openBZIssue(reportURL)}
+                    component="a">
+                      {_("Report issue")}
+                  </Button>}
                   <Button variant="secondary" onClick={exitGui}>
                       {isBootIso ? _("Reboot") : _("Quit")}
                   </Button>
               </>
           }>
-            <TextContent>
-                <Text component={TextVariants.p}>
-                    {exception.message}
-                </Text>
-            </TextContent>
+            <Form>
+                <FormGroup label={_("Error details")}>
+                    <TextContent>
+                        <Text component={TextVariants.p}>
+                            {exception.name + ": " + exception.message}
+                        </Text>
+                    </TextContent>
+                </FormGroup>
+                <FormGroup
+                  label={_("Log")}
+                >
+                    <TextArea
+                      value={logContent}
+                      onChange={setLogContent}
+                      aria-label="review-attached-log"
```

Comment:
aria-label should be a normal text like "Review attached log"
In this case there is already the label just above so you just need to add an id in the TextArea, and the 'fieldId' property in the FormGroup

Reply:
Thank you, updated.